x/wasm/keeper/querier.go

```diff
@@ -311,3 +311,48 @@ func (q GrpcQuerier) PinnedCodes(c context.Context, req *types.QueryPinnedCodesR
 		Pagination: pageRes,
 	}, nil
 }
+
+func (q GrpcQuerier) InactiveContracts(c context.Context, req *lbmtypes.QueryInactiveContractsRequest) (*lbmtypes.QueryInactiveContractsResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+	ctx := sdk.UnwrapSDKContext(c)
+
+	addresses := make([]string, 0)
+	prefixStore := prefix.NewStore(ctx.KVStore(q.storeKey), types.InactiveContractPrefix)
+	pageRes, err := query.FilteredPaginate(prefixStore, req.Pagination, func(key []byte, value []byte, accumulate bool) (bool, error) {
+		if accumulate {
+			contractAddress := sdk.AccAddress(value)
+			addresses = append(addresses, contractAddress.String())
+		}
+		return true, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return &lbmtypes.QueryInactiveContractsResponse{
+		Addresses:  addresses,
+		Pagination: pageRes,
+	}, nil
+}
+
+func (q GrpcQuerier) InactiveContract(c context.Context, req *lbmtypes.QueryInactiveContractRequest) (*lbmtypes.QueryInactiveContractResponse, error) {
```

Comment:
I think `isInactiveContract` is better name?

Reply:
It's a little strange. But I wanted to provide the singular variation of this query of `InactiveContracts`. So I decided using this name.